docs/product/specs/permissions-revamp.md

```diff
@@ -0,0 +1,115 @@
+* Admins should be able to create new roles on the database server and associate them with the database.
+  - This will require the role used by the admin to have CREATEROLE privilege.
+* Future considerations:
+  - Roles exist on the database server and not under a database.
+  - However, inorder to simplify the user flow, we are not introducing a UX dedicated for database servers.
+  - We can explore this option in the future, based on user feedback or product strategy discussions.
+
+### Access control options
+* Admins should be able to provide Mathesar users access to each database configured in Mathesar.
+* Each Mathesar user, including admins, should only be associated with a single role per database.
+  - Admins should be able to update this user-to-role configuration.
+* All Mathesar admins should be able to access all databases configured in Mathesar. Admins cannot be removed from a database.
+  - When a Mathesar admin is newly added into the product, they will automatically be provided the 'default role' for all databases.
+  - The concept of a 'default role' is explained further below in this document.
+* The role for admins can be changed.
+  - This can be restricted to reduce scope and we could mention that admins will always assume the default role.
+  - The challenges are mentioned in further below in the 'Dealing with Ownership and Collaboration issues' section of this document.
+* Standard Mathesar users can be added/removed from a database.
+* Standard Mathesar users will not be able to access a database without being provided explicit access by a Mathesar admin.
+* The permissions for DB objects will be determined by the priviliges of the role the user is associated with.
+* For Mathesar specific objects, we will have a setting per user per database that allows the user to 'read' or 'read-write'.
+  - This applies for explorations and metadata such as table and column settings.
+  - Future consideration:
+    - We can allow an additional option 'Infer from DB privileges' that will automatically identify access to each Mathesar-object based on the privileges on the underlying DB objects.
+
+### Schema level access control options
+* If a user has ownership access (or appropriate GRANT priviliges) to a schema based on their underlying role, they should be able to grant/revoke access to that schema to other roles configured in Mathesar.
+* While granting access to a role, the user can choose access control at three different levels.
+  - manage: USAGE, GRANT, SELECT, INSERT, UPDATE, DELETE, SET and ALTER
+  - edit: USAGE, SELECT, INSERT, UPDATE, DELETE, SET and ALTER
+  - view: USAGE, SELECT
+* This setting will iteratively apply to all objects within the schema.
+* For schemas created using the Mathesar UI, we will automatically grant view access to all configured roles, manage access to the 'default role', and we will provide access accordingly for auto-generated roles.
+* This will not be present for the 'public schema'.
```

Comment:
We should discuss in a call the changes around permissions granted to the `PUBLIC` group in PG 16.

Reply:
Mentioned in https://github.com/mathesar-foundation/mathesar-wiki/commit/2ad0d7262212be9b5897deda8b7962e99d2f7191.